### ISSUE TITLE
Metadata plus newline breaks the reader

### ISSUE THREAD

**Issue by arichiardi** [anmonteiro/lumo#93]
Hello Antonio,

I have just discovered a bug in the `ns` form, not sure it applies upstream or not, I haven't checked and I apologize if it is.

Basically this throws an error:

```
foo.bar=> (ns ^{:author "Andrea Richiardi"}
            foo.bar
            (:require [clojure.walk :as walk]))
#error {:message EOF, :data {:type :reader-exception}}
foo.bar=> #js {}
foo.bar=> WARNING: Use of undeclared Var foo.bar/walk at line 1 
nil
#error {:message Unmatched delimiter ), :data {:type :reader-exception}}
```

While this not:

```
foo.bar=> (ns ^{:author "Andrea Richiardi"} foo.bar
            (:require [clojure.walk :as walk]))
#_=> nil
```

And this is again:

```
foo.bar=> (ns ^{:author "Andrea Richiardi"}
            foo.bar)
#error {:message EOF, :data {:type :reader-exception}}
foo.bar=> #js {}
#error {:message Unmatched delimiter ), :data {:type :reader-exception}}
```

Thanks!

**Comment by anmonteiro** [comment]
Looks like it might be a Lumo issue, not upstream

**Comment by arichiardi** [comment]
Yes also this does not detect that a closing paren is missing and does not wrap:

```
cljs.user=> (ns ^{:author "tart"}
#error {:message EOF, :data {:type :reader-exception}}
```

**Comment by anmonteiro** [comment]
FWIW this also happens in Planck, but gives no error

**Comment by arichiardi** [comment]
The problem appears for `def` as well:

```
(def ^{:doc "Take the third element using nth"}
  third #(nth % 2))
#error {:message EOF, :data {:type :reader-exception}}
speed-of-light.aws=> #object[speed_of_light$aws$third "function speed_of_light$aws$third(p1__31_SHARP_){
return cljs.core.nth.call(null,p1__31_SHARP_,(2));
}"]
#object[Function "function (p1__34_SHARP_){
return cljs.core.nth.call(null,p1__34_SHARP_,(2));
}"]
#error {:message Unmatched delimiter ), :data {:type :reader-exception}}
```

I'll change the title.

**Comment by anmonteiro** [comment]
Not reproducible anymore (looks fixed in 1.7).

**Comment by anmonteiro** [comment]
This would be a candidate for an integration test cc @arichiardi
